augur/sequence_traits.py

```diff
@@ -163,7 +164,15 @@ def read_in_features(drm_file):
 
     mutPositions = defaultdict(list)
 
-    df = pd.read_csv(drm_file, sep='\t' if drm_file.endswith('.tsv') else ',')
+    df = pd.read_csv(drm_file, sep=None)
+
+    # Validate column names.
+    observed_columns = set(df.columns)
+    required_columns = {"GENE", "SITE", "ALT", "FEATURE"}
+    optional_columns = {"DISPLAY_NAME"}
+    if observed_columns - optional_columns != required_columns:
+        raise AugurError(f"{drm_file} has columns {sorted(observed_columns)}, but expected {sorted(required_columns)} with optional {sorted(optional_columns)}.")
+
     for mi, m in df.iterrows():
         pos = m.SITE-1 #put in python numbering
         gene = m.GENE if hasattr(m, 'GENE') else 'nuc'

```

Comment:
This `hasattr()` guards against inputs without a `GENE` column, yeah? (by defaulting the "gene" to `nuc`)